server.go

```diff
@@ -2119,7 +2119,11 @@ func ClientSupportedCompressors(ctx context.Context) ([]string, error) {
 		return nil, fmt.Errorf("failed to fetch the stream from the given context %v", ctx)
 	}
 
-	return strings.Split(stream.ClientAdvertisedCompressors(), ","), nil
+	values := strings.Split(stream.ClientAdvertisedCompressors(), ",")
```

Comment:
It seems `ClientAdvertisedCompressors()` is always used with a `strings.Split(",")`.  Can we make that return `[]string` instead and do the split/trim there, please?

Reply:
@dfawley I commited the changes you requested. Could you review again 🙏